manifests/init.pp

```diff
@@ -168,7 +184,31 @@
   Hash                                                $dropin_files = {},
   Hash                                                $udev_rules = {},
 ) {
-  create_resources('systemd::service_limits', $service_limits)
+  $service_limits.each |String[1] $service_limit, Hash[String[1], Any] $service_limit_data| {
```

Comment:
This feels overly strict. Is it really needed to set data types in these lines? IMHO you don't get any additional type safety while it is longer to read and also means you need to update it in 2 places.

Reply:
cleaned it up